imu.c

```diff
@@ -0,0 +1,534 @@
+        config[ACCEL].cfg.acc.odr = BMI2_ACC_ODR_200HZ;
+
+        /* Gravity range of the sensor (+/- 2G, 4G, 8G, 16G). */
+        config[ACCEL].cfg.acc.range = BMI2_ACC_RANGE_2G;
+
+        /* The bandwidth parameter is used to configure the number of sensor samples that are averaged
+         * if it is set to 2, then 2^(bandwidth parameter) samples
+         * are averaged, resulting in 4 averaged samples.
+         * Note1 : For more information, refer the datasheet.
+         * Note2 : A higher number of averaged samples will result in a lower noise level of the signal, but
+         * this has an adverse effect on the power consumed.
+         */
+        config[ACCEL].cfg.acc.bwp = BMI2_ACC_NORMAL_AVG4;
+
+        /* Enable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         * For more info refer datasheet.
+         */
+        config[ACCEL].cfg.acc.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* The user can change the following configuration parameters according to their requirement. */
+        /* Set Output Data Rate */
+        config[GYRO].cfg.gyr.odr = BMI2_GYR_ODR_200HZ;
+
+        /* Gyroscope Angular Rate Measurement Range.By default the range is 2000dps. */
+        config[GYRO].cfg.gyr.range = BMI2_GYR_RANGE_2000;
+
+        /* Gyroscope bandwidth parameters. By default the gyro bandwidth is in normal mode. */
+        config[GYRO].cfg.gyr.bwp = BMI2_GYR_NORMAL_MODE;
+
+        /* Enable/Disable the noise performance mode for precision yaw rate sensing
+         * There are two modes
+         *  0 -> Ultra low power mode(Default)
+         *  1 -> High performance mode
+         */
+        config[GYRO].cfg.gyr.noise_perf = BMI2_POWER_OPT_MODE;
+
+        /* Enable/Disable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         */
+        config[GYRO].cfg.gyr.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* Set the accel and gyro configurations. */
+        rslt = bmi2_set_sensor_config(config, 2, bmi);
+        bmi2_print_error_code(rslt);
+    }
+
+    return rslt;
+}
+
+/*!
+ * @brief This function converts lsb to meter per second squared for 16 bit accelerometer at
+ * range 2G, 4G, 8G or 16G.
+ */
+static float lsb_to_mps2(int16_t val, float g_range, uint8_t bit_width)
+{
+    double power = 2;
+
+    float half_scale = (float)((pow((double)power, (double)bit_width) / 2.0f));
+
+    return (GRAVITY_EARTH * val * g_range) / half_scale;
+}
+
+/*!
+ * @brief This function converts lsb to degree per second for 16 bit gyro at
+ * range 125, 250, 500, 1000 or 2000dps.
+ */
+static float lsb_to_dps(int16_t val, float dps, uint8_t bit_width)
```

Comment:
all static methods at the top of the file, this will let you remove the empty forward declarations you've created.